src/test/java/org/broadinstitute/barclay/argparser/CommandLinePluginUnitTest.java

```diff
@@ -453,4 +453,67 @@ public void testPluginArgumentNameCollision(){
                 Collections.emptySet());
     }
 
+    // Used to test the default implementation of includePluginClass
+    public static class DescriptorThatDoesntOverrideIncludePluginClass extends CommandLinePluginDescriptor<TestPluginBase> {
+        @Override
+        public List<String> getPackageNames() {
+            return null;
+        }
+
+        @Override
+        public Class<?> getPluginBaseClass() {
+            return TestPluginBase.class;
+        }
+
+        @Override
+        public TestPluginBase createInstanceForPlugin(Class<?> pluginClass) throws IllegalAccessException, InstantiationException {
+            return null;
+        }
+
+        @Override
+        public boolean isDependentArgumentAllowed(Class<?> predecessorClass) {
+            return false;
+        }
+
+        @Override
+        public void validateAndResolvePlugins() throws CommandLineException {
+
+        }
+
+        @Override
+        public List<TestPluginBase> getDefaultInstances() {
+            return null;
+        }
+
+        @Override
+        public List<TestPluginBase> getResolvedInstances() {
+            return null;
+        }
+
+        @Override
+        public Set<String> getAllowedValuesForDescriptorHelp(String longArgName) {
+            return null;
+        }
+
+        @Override
+        public Class<?> getClassForPluginHelp(String pluginName) {
+            return null;
+        }
+    }
+
+    @DataProvider(name="includePluginClassTests")
+    Object[][] getIncludePluginClassTests() {
+        return new Object[][] {
+                { TestPluginBase.class, true },
+                { TestPluginWithOptionalArg.class, true },
+                { Integer.class, false }
+        };
+    }
+
+    @Test(dataProvider = "includePluginClassTests")
+    public void testDefaultIncludePluginClass(final Class<?> clazz, final boolean expectedInclusion) {
+        final DescriptorThatDoesntOverrideIncludePluginClass descriptor = new DescriptorThatDoesntOverrideIncludePluginClass();
```

Comment:
You can use a mocked class instead of a full implementation of the plugin...

Reply:
Excellent idea. It was not quite as straightforward as I thought though. Still, its a lot less code.